File: NotesServer/Models/CollaborationDatabase.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace NotesServer.Models
{
    public class CollaborationDatabase
    {
        [Key]
        public string Id { get; set; } = Guid.NewGuid().ToString();
        
        [Required]
        public string UserId { get; set; } = string.Empty;
        
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        
        [Required]
        public string DatabaseName { get; set; } = string.Empty;
        
        [Required]
        public string ConnectionString { get; set; } = string.Empty;
    }
} 

File: lib/models/collaboration_database.dart
class CollaborationDatabase {
  final String id;
  final String userId;
  final DateTime createdAt;
  final String databaseName;
  final String connectionString;

  CollaborationDatabase({
    required this.id,
    required this.userId,
    required this.createdAt,
    required this.databaseName,
    required this.connectionString,
  });

  factory CollaborationDatabase.fromJson(Map<String, dynamic> json) {
    return CollaborationDatabase(
      id: json['id'],
      userId: json['userId'],
      createdAt: DateTime.parse(json['createdAt']),
      databaseName: json['databaseName'],
      connectionString: json['connectionString'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'userId': userId,
      'createdAt': createdAt.toIso8601String(),
      'databaseName': databaseName,
      'connectionString': connectionString,
    };
  }
} 